datadog-slos-slo-handler/src/datadog_slos_slo/handlers.py

```diff
@@ -0,0 +1,234 @@
+        slo_id = model.Id
+        try:
+            slo = api_instance.get_slo(slo_id)
+        except ApiException as e:
+            LOG.error("Exception when calling SLOApi->get_slo: %s\n", e)
+            return ProgressEvent(
+                status=OperationStatus.FAILED, resourceModel=model,
+                message=f"Error getting monitor: {e}"
+            )
+    model.Created = slo.data.created_at
+    model.Modified = slo.data.modified_at
+    model.Description = slo.data.description
+    model.Name = slo.data.name
+    model.Tags = slo.data.tags
+    model.Type = slo.data.type.value
+    if slo.data.creator:
+        model.Creator = Creator(Name=slo.data.creator.name, Email=slo.data.creator.email,
+                                Handle=slo.data.creator.handle)
+
+    if slo.data.type.value == "monitor":
+        if hasattr(slo.data, 'groups'):
+            model.Groups = slo.data.groups
+        model.MonitorIds = slo.data.monitor_ids
+    elif slo.data.type.value == "metric":
+        model.Query = Query(
+            Denominator=slo.data.query.denominator,
+            Numerator=slo.data.query.numerator
+        )
+    thresholds = []
+    for th in slo.data.thresholds:
+        thresholds.append(Threshold(
+            Target=th.target if hasattr(th, "target") else None,
+            TargetDisplay=th.target_display if hasattr(th, "target_display") else None,
+            Timeframe=th.timeframe.value if hasattr(th, "timeframe") else None,
+            Warning=th.warning if hasattr(th, "warning") else None,
+            WarningDisplay=th.warning_display if hasattr(th,"warning_display") else None
+        ))
+    model.Thresholds = thresholds
+
+    model.Id = slo.data.id
+
+    return ProgressEvent(
+        status=OperationStatus.SUCCESS,
+        resourceModel=model,
+    )
+
+
+@resource.handler(Action.UPDATE)
+def update_handler(
+        session: Optional[SessionProxy],
+        request: ResourceHandlerRequest,
+        callback_context: MutableMapping[str, Any],
+) -> ProgressEvent:
+    LOG.info("Starting %s Update Handler", TYPE_NAME)
+    model = request.desiredResourceState
+    thresholds = build_slo_thresholds_from_model(model)
+
+    slo = ApiSLO(name=model.Name, type=ApiSLOType(model.Type), thresholds=thresholds)
+    if model.Description is not None:
+        slo.description = model.Description
+    if model.Groups is not None:
+        slo.groups = model.Groups
+    if model.MonitorIds is not None:
+        slo.monitor_ids = model.MonitorIds
+    if model.Query is not None:
+        slo.query = ApiSLOQuery(numerator=model.Query.Numerator,
+                                denominator=model.Query.Denominator)
+    if model.Tags is not None:
+        slo.tags = model.Tags
+
+    with v1_client(
+            model.DatadogCredentials.ApiKey,
+            model.DatadogCredentials.ApplicationKey,
+            model.DatadogCredentials.ApiURL or "https://api.datadoghq.com",
```

Comment:
you can remove the `or "https://....` here. The default is now included when you use the latest version of the common package

Reply:
With the update to TypeConfiguration this has been changed